pkg/registry/apis/datasource/sub_resource.go

```diff
@@ -0,0 +1,79 @@
+package datasource
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	"github.com/grafana/grafana/pkg/plugins/httpresponsesender"
+)
+
+type subResourceREST struct {
+	builder *DSAPIBuilder
+}
+
+var _ = rest.Connecter(&subResourceREST{})
+
+func (r *subResourceREST) New() runtime.Object {
+	return &metav1.Status{}
+}
+
+func (r *subResourceREST) Destroy() {
+}
+
+func (r *subResourceREST) ConnectMethods() []string {
+	// All for now??? ideally we have a schema for resource and limit this
+	return []string{
+		http.MethodGet,
+		http.MethodHead,
+		http.MethodPost,
+		http.MethodPut,
+		http.MethodPatch,
+		http.MethodDelete,
+		http.MethodOptions,
+	}
+}
+
+func (r *subResourceREST) NewConnectOptions() (runtime.Object, bool, string) {
+	return nil, true, ""
```

Comment:
@andresmgot -- long subpaths should work now.  The second true parameter appears to register everything after the slash also.

Welcome to the mysterious world of kubernetes options!